### DISCUSSION TITLE
0.13.0 panics on liberica jre (armv7)

### DISCUSSION THREAD

**Discussion by MichaelBitard** [AsamK/signal-cli#1478] [Q&A] [answered]
Hi, I'm facing an issue I'm not sure how I can debug it.
Since the upgrade to java 21 I cannot use debian's default jre (17) and oracle dropped support for a lot of architectures :disappointed:

I tried the liberica JRE, which has binaries for the architectures I'm using (RPI 3 for example)

Anyway, when launching signal-cli as a system daemon `signal-cli --config /path -u +33XXXXXX daemon --system` It fails with the message: "failed to initiate panic, error 9"

It does not seem to be a SIGKILL error (9) so I don't know what I can do from there.

EDIT:
I followed jre installation from here and it seems I have java correctly installed: https://docs.bell-sw.com/liberica-jdk/21.0.2b14/general/install-guide/#linux

**Comment by m-ueberall** [comment]
Just a hunch–did you check that the `libsignal_jni.so` library you're using is compatible with your version of Debian? The below should display the required minimum glibc version after you extract it from the `libsignal-client-a.bb.c.jar`:
```
objdump -T ./libsignal_jni.so | grep GLIBC | sed -e 's/.* (GLIBC_//g' -e 's/).*//g' | sort -V | uniq | tail -1
```

**Reply by MichaelBitard** [reply]
it was compatible with the previous version, the v0.36.1. I'll check if it is still compatible with the latest (v0.39.2)

**Reply by MichaelBitard** [reply]
The glibc version did not change between the v0.36.1 and v0.39.2 (2.28)

**Comment by m-ueberall** [comment]
Maybe your problem is related to another library? Could you repeat the above process with `org/sqlite/native/Linux/armv7/libsqlitejdbc.so` found in `sqlite-jdbc-3.45.1.0.jar` (see [here](https://github.com/exquo/signal-libs-build/issues/17#issuecomment-1964325241))? (To ensure that this is the one in use, you could temporarily remove all other variants in this archive.)

**Reply by MichaelBitard** [reply]
This is still the same glibc version 2.28?. I tried what you said and removed all the other variants, and I still get the same error.

I have now access to a simple rpi3 with docker on it, I should be able to test more rapidly

**Comment by MichaelBitard** [comment]
I have better logs now:

```
2024-07-17T07:43:17.831Z [main] DEBUG org.asamk.signal.util.IOUtils - XDG_RUNTIME_DIR not set, falling back to temp dir
2024-07-17T07:43:17.917Z [main] DEBUG org.asamk.signal.App - Starting signal-cli 0.13.0
2024-07-17T07:43:18.197Z [main] INFO  LibSignal - [libsignal]: rust/bridge/jni/src/logging.rs:158: Initializing libsignal version:0.40.1
2024-07-17T07:43:18.203Z [main] ERROR LibSignal - [libsignal]: /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/log-panics-2.1.0/src/lib.rs:130: thread '<unnamed>' panicked at 'Java classes were not preloaded': rust/bridge/shared/src/jni/mod.rs:751
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: Java_org_signal_libsignal_internal_Native_UuidCiphertext_1CheckValidContents

fatal runtime error: failed to initiate panic, error 9
Aborted (core dumped)
```

**Reply by MichaelBitard** [reply, marked as answer]
I updated signal-cli to 0.13.4 and it seems to work now, I'll try to send/receive message and if it works I'll close the discussion
